independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -981,7 +1009,7 @@ void processAnnotation(AnnotationInstance annotation,
                 qualifiers.add(qualifierAnnotation);
             }
             // Treat the case when an additional bean defining annotation that is also a qualifier declares the default scope
```

Comment:
Interesting. The code for producer methods/fields doesn't have this IIRC?

Reply:
Ah, didn't notice this. Yes, so this was actually used to solve the problem we have in https://github.com/quarkusio/quarkus/pull/27616 for class beans...